### PR TITLE
rancher-cli: Add version 2.4.0

### PR DESCRIPTION
**Pull request by reicheltp**
The Rancher Command Line Interface (CLI) is a unified tool for interacting with your Rancher Server.

For usage information see: rancher.com/docs/rancher/v2.x/en/cli

The Rancher CLI is widely used with over 1 million total downloads.

### CODE REVIEW

File: bucket/rancher-cli.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        "64bit": {
+            "url": "https://github.com/rancher/cli/releases/download/v2.4.0/rancher-windows-amd64-v2.4.0.zip",
+            "hash": "0a5414a85fe58da10ac3d547dc9486d2176a867dbc589b60bbb77cb95182e859"
+        },
+        "32bit": {
+            "url": "https://github.com/rancher/cli/releases/download/v2.4.0/rancher-windows-386-v2.4.0.zip",
+            "hash": "ea948577e3b5a92211859903e321f46ca6548067e928d8de7f7415463e9eb3a6"
+        }
+    },
+    "pre_install": "mv $original_dir/rancher-v$version/* $original_dir/",
```
Comment (Ash258) [inline bucket/rancher-cli.json:16]:
```suggestion
    "extract_dir": "rancher-v2.4.0",
```

File: bucket/rancher-cli.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            "64bit": {
+                "url": "https://github.com/rancher/cli/releases/download/v$version/rancher-windows-amd64-v$version.zip"
+            },
+            "32bit": {
+                "url": "https://github.com/rancher/cli/releases/download/v$version/rancher-windows-386-v$version.zip"
+            }
+        },
+        "hash": {
+            "url": "$baseurl/sha256sum.txt"
+        }
```
Comment (Ash258) [inline bucket/rancher-cli.json:30]:
```suggestion
        },
        "extract_dir": "rancher-v$version"
```

File: bucket/rancher-cli.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+    "version": "2.4.0",
+    "description": "The Rancher Command Line Interface (CLI) is a unified tool for interacting with your Rancher Server.",
```
Comment (Ash258) [inline bucket/rancher-cli.json:3]:
```suggestion
    "description": "Unified tool for interacting with Rancher Server",
```

**Comment by reicheltp** [comment]
@Ash258 Thanks for the suggestion. Didn't know that you can patch `extract_dir` with the autoupdate function.

**Comment by reicheltp** [comment]
@Ash258 Is there anything left to do, to get this merged?

**Pull request merged**